qiskit_nature/operators/second_quantization/fermionic_op.py

```diff
@@ -358,6 +371,45 @@ def _to_sparse_label(label):
             filter(None, (label_transformation[c].format(i=i) for i, c in enumerate(label)))
         )
 
+    def to_normal_order(self) -> "FermionicOp":
+        """Convert to the normal order. This is not usual normal order.
+        It returns the operator with normal order but equivalent to the input operator.
+        For example, in usual :math:`\\colon c_0 c_0^\\dagger\\colon = -c_0^\\dagger c_0`
+        where :math:`c_0` is an annihilation operator,
+        but this method returns :math:`-c_0^\\dagger c_0 + 1`.
+        The returned operator is a sparse label mode.
+        """
```

Comment:
I don't quite understand what you are trying to say here.. Shouldn't this function always perform the computation according to:
```
c_i c_j^\dagger = \delta_ij - c_j^\dagger c_i
```
where `\delta_ij` is the Kronecker delta (i.e. `delta = 1 iff i == j else 0`)

According to the following test this is indeed the case:
```python
>>> print(FermionicOp("E_0", register_length=4).to_normal_order())
   * (1+0j)    # <- notice this weird formatting edge case
+ +_0 -_0 * (-1+0j)
>>> print(FermionicOp("-_0 +_1", register_length=4).to_normal_order())
+_1 -_0 * (-1+0j)
```

Reply:
You are right. In the usual normal order, `:c_i c_j†: = - c_j† c_i`, my intention is to note that this method behaves differently than usual.
(And I'll notice the weird formatting.. Thanks. I'll fix it with truncation in the next PR.)

Reply:
What I meant is that I don't understand how this method is different than the usual normal oder?

Reply:
I see.
Usual: `: c_0 c_0^†: = -c_0^† c_0.`
This method: `: c_0 c_0^†: = 1 - c_0^† c_0.`

Reply:
No but this is exactly what I am saying. Because these are *fermionic* operators, the definition is:
```
:c_i c_j^†: = \delta_ij - c_j^† c_i
```
This means that the usual case is:
* `:c_i c_j^†: = - c_j^† c_i` if and only if `i != j`
* `:c_i c_i^†: = 1 - c_i^† c_i` if and only if `i == j`

Thus, this method does exactly what is expected from it.

Reply:
Your reference wikipedia said
<img width="140" alt="image" src="https://user-images.githubusercontent.com/6814928/127996269-4330daf3-b3db-444f-837c-3f18de2cbdf3.png">

Reply:
https://physics.stackexchange.com/questions/149419/normal-ordering-for-a-two-fermion-case

Reply:
That is the single fermions section. You need to consider the multiple fermions case, which is this:
![screenshot_1627985202](https://user-images.githubusercontent.com/21973473/127998226-7d2288ef-5672-46b1-81d7-b35a3510aa4f.png)
The most important part being the very last line on this screenshot

Reply:
The last line is just the commutation relation, not the definition of the normal order.

Reply:
<img width="635" alt="image" src="https://user-images.githubusercontent.com/6814928/128016078-482c675c-2ffb-4a0a-9e9d-b1325778039b.png">

Reply:
If we adopt your definition, 
```
:f_2 f_1^\dagger f_2^\dagger: = -f_1^\dagger f_2 f_2^\dagger = - f_1^\dagger (1 - f_2^\dagger f_2) = - f_1^\dagger + f_1^\dagger f_2^\dagger f_2.
```
However Wikipedia said `:f_2 f_1^\dagger f_2^\dagger: = f_1^\dagger f_2^\dagger f_2`.

Reply:
And, the case for single fermion means we only have single register.

Reply:
In the first place, the normal ordered product is used to remove divergence of vacuum expectation value in field theory, so it can't be an equivalent operator.

Reply:
@ikkoham thanks for explaining this. Please add a reference to the docstring for future reference so that this is clear. I think that the misunderstanding here is that @ikkoham is talking about the definition of normal order (that is indeed right) and @mrossinek is taking about how to do an algebraic transformation of a random ordered operator to a normal ordered operator (that needs to take into account the commutation rules and the above definition is also right). The important is that the code does the transformation to normal order in the proper way. Thanks!

Reply:
Thanks for clarifying @pbark! In that case the docstring should simply reflect that the operator gets transformed according to the anticommutation rules :+1: